functions/__z.fish

```diff
@@ -68,9 +68,6 @@ function __z -d "Jump to a recent directory."
                         printf "%-10s %s\n", matches[x], x | cmd
                     }
                 }
-                if( common ) {
-                    printf "%-10s %s\n", "common:", common > "/dev/stderr"
-                }
```

Comment:
The fact that `z -l query` prints the common prefix looks like a nice feature. How about we keep it and use this alternative fix:

```diff
diff --git a/functions/__z_complete.fish b/functions/__z_complete.fish
index 51ec5f8..fe4f641 100644
--- a/functions/__z_complete.fish
+++ b/functions/__z_complete.fish
@@ -3,8 +3,8 @@ function __z_complete -d "add completions"
         printf "%s\n" (string replace -r '\|.*' '' < $Z_DATA)
     end
 
-    complete -c $Z_CMD -a "(__z -l | string replace -r '^\\S*\\s*' '')" -f -k
-    complete -c $ZO_CMD -a "(__z -l | string replace -r '^\\S*\\s*' '')" -f -k
+    complete -c $Z_CMD -a "(__z -l 2>/dev/null | string replace -r '^\\S*\\s*' '')" -f -k
+    complete -c $ZO_CMD -a "(__z -l 2>/dev/null | string replace -r '^\\S*\\s*' '')" -f -k
 
     complete -c $Z_CMD -s c -l clean -d "Cleans out $Z_DATA"
     complete -c $Z_CMD -s e -l echo -d "Prints best match, no cd"
```

Reply:
Yeah the thing is, this still retains the pattern of outputting the common path to stderr for `list`. Which is the core of the problem. Stderr is for errors or warnings.

If you take the list, you could extract the common path from that separately. For example

![image](https://user-images.githubusercontent.com/7355199/162489607-680ea3e6-4aa6-4953-a81b-ef77a0d05d95.png)

If finding a common prefix is a very common usecase and you want to make it simpler for your users, I would provide it under a separate flag, say `--common`. That way if a user wants the common prefix, they can have it, if they want the list, that's possible as well. And if they want to retain the old behaviour, they could create an alias alias that calls `--common` and `--list` in sequence.

Everything would then be possible without messing up the prompt.

Reply:
I think the core of my point is, it seems more in line with the unix philosophy to have list output a list. And if finding a common prefix is useful, having that as a separate command seems like the most composable solution. That would allow choice and composability without sacrificing us following the standards (i.e. using stderr vs stdout).